### PR TITLE
feat(cse): support a new Microservice engine resource

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Support a new resource to manage the microservice resources (CSE 2.0+).

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support a new microservice engine resource.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
Pending
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cse' TESTARGS='-run=TestAccMicroserviceEngine_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cse -v -run=TestAccMicroserviceEngine_basic -timeout 360m -parallel 4
=== RUN   TestAccMicroserviceEngine_basic
=== PAUSE TestAccMicroserviceEngine_basic
=== CONT  TestAccMicroserviceEngine_basic
--- PASS: TestAccMicroserviceEngine_basic (967.96s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cse 968.076s
```

File: huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+			},
+			"auth_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"RBAC", "NONE",
+				}, false),
+			},
+			"auth_cred": {
```
Comment (hmbig2) [inline huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go]: only one item, and one attr,  change the type to schema.TypeString

File: docs/resources/cse_microservice_engine.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+* `availability_zones` - (Required, List, ForceNew) Specifies the list of availability zone.
+  Changing this will create a new engine.
+
+* `network_id` - (Required, String, ForceNew) Specifies the network ID of the subnet to which the dedicated microservice
+  engine belongs. Changing this will create a new engine.
+
+* `auth_type` - (Required, String, ForceNew) Specifies the authentication method for the dedicated microservice engine.
+  Changing this will create a new engine.
+  + **true**: Enable security authentication.
```
Comment (hmbig2) [inline docs/resources/cse_microservice_engine.md]: in the schema , enum is `RBAC` and `NONE`

Comment (hmbig2) [inline docs/resources/cse_microservice_engine.md:51]: in the schema, enum is `RBAC` and `NONE`

File: docs/resources/cse_microservice_engine.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  + **false**: Disable security authentication.
+    After security authentication is disabled, all users who use the engine can use the engine without using the account
+    and password, and have the same operation permissions on all services.
+
+* `auth_cred` - (Optional, List, ForceNew) Specifies the authentication configuration.
+  Required if `auth_type` is **true**. The [object](#microservice_auth_cred) structure is documented below.
+  Changing this will create a new engine.
+
+* `description` - (Optional, String, ForceNew) Specifies the description of the dedicated microservice engine.
+  The description can contian a maximum of `128` characters.
```
Comment (hmbig2) [inline docs/resources/cse_microservice_engine.md]: in the schema, is 0~255

File: huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	client, err := conf.CseV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CSE v2 client: %s", err)
+	}
+
+	resp, err := engines.Get(client, d.Id(), common.GetEnterpriseProjectID(d, conf))
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Microservice engine")
```
Comment (hmbig2) [inline huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go]: pls use parseEngineJobError to wrap the error

Comment (Lance52259) [reply huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go]: done.

File: docs/resources/cse_microservice_engine.md (hunk 3 above)
Comment (Lance52259) [reply docs/resources/cse_microservice_engine.md]: done.

File: docs/resources/cse_microservice_engine.md (hunk 2 above)
Comment (Lance52259) [reply docs/resources/cse_microservice_engine.md]: done.

File: huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cse' TESTARGS='-run=TestAccMicroserviceEngine_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cse -v -run=TestAccMicroserviceEngine_basic -timeout 360m -parallel 4
=== RUN   TestAccMicroserviceEngine_basic
=== PAUSE TestAccMicroserviceEngine_basic
=== CONT  TestAccMicroserviceEngine_basic
--- PASS: TestAccMicroserviceEngine_basic (854.63s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cse       854.973s
```

File: huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+			},
+			"auth_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"RBAC", "NONE",
+				}, false),
+			},
+			"admin_pass": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go:83]: missing Sensitive？

File: huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
```
Comment (ShiChangkuo) [inline huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go]: Computed is useless.

File: huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+		d.Set("description", resp.Description),
+		d.Set("eip_id", resp.Reference.PublicIpId),
+		d.Set("extend_params", resp.Reference.Inputs),
+	)
+	// Attributes
+	if resp.Reference.ServiceLimit != "" {
+		limit, err := strconv.Atoi(resp.Reference.ServiceLimit)
+		if err != nil {
+			mErr = multierror.Append(mErr, fmt.Errorf("unable to parsing service limit (%#v)",
+				resp.Reference.ServiceLimit))
```
Comment (ShiChangkuo) [inline huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go]: can we add it into diags with warnning level or just ignore it?

Comment (Lance52259) [reply huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go]: done.

File: huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go (hunk 6 above)
Comment (Lance52259) [reply huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go]: sure, it's unnecessary.

File: huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go (hunk 5 above)
Comment (Lance52259) [reply huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go:83]: done.

**Pull request merged**